mobile/library/common/jni/jni_interface.cc

```diff
@@ -1272,13 +1272,25 @@ void configureBuilder(JNIEnv* env, jstring grpc_stats_domain, jboolean admin_int
 
   std::vector<std::string> hostnames = javaObjectArrayToStringVector(env, dns_preresolve_hostnames);
   builder.addDnsPreresolveHostnames(hostnames);
-  builder.addRtdsLayer(getCppString(env, rtds_layer_name), rtds_timeout_seconds);
-  builder.setNodeId(getCppString(env, node_id));
-  builder.setNodeLocality(getCppString(env, node_region), getCppString(env, node_zone),
-                          getCppString(env, node_sub_zone));
-  builder.setAggregatedDiscoveryService(getCppString(env, ads_address), ads_port,
-                                        getCppString(env, ads_token), ads_token_lifetime,
-                                        getCppString(env, ads_root_certs));
+  std::string native_rtds_layer_name = getCppString(env, rtds_layer_name);
+  if (!native_rtds_layer_name.empty()) {
+    builder.addRtdsLayer(native_rtds_layer_name, rtds_timeout_seconds);
+  }
+  std::string native_node_id = getCppString(env, node_id);
+  if (!native_node_id.empty()) {
+    builder.setNodeId(native_node_id);
```

Comment:
not sure how it is supposed to work but should we put all of these builder calls behind their `if` conditional checks as they are now + `if (!native_rtds_layer_name.empty())` ?

Reply:
It was still compiling because I had only added `#ifdef` checks to the implementation (`.cc`) files and not the header files.

I removed all the build system changes to reduce the scope of this PR.

Reply:
Yea, I guess my question was so whether we need to set attributes such as native_node_id etc in case `native_rtds_layer_name.empty() == true` as it seems to me that we use node attributes in the context of the runtime layer only. My proposal was to put updates of all of the node attributes behind `native_rtds_layer_name.empty() == true` check but I do not think that it's a requirement.

Curious of your thoughts on this.

Reply:
Yeah I had similar feedback in the initial PR, but ultimately since these can now be set independently, I'd rather keep behavior closer to what the C++ engine builder does.